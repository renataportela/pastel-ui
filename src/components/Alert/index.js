export { default, ALERT_COLORS } from './Alert'