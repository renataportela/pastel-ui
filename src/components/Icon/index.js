export { default, iconNames } from './Icon';
export { default as Svg } from './Svg';