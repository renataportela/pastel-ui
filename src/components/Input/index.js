export { default, INPUT_PROPS } from './Input'
export { inputStyles } from './ElementStyle'
