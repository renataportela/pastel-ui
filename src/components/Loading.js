import React from 'react'
import { colors, Icon, Overlay } from '~/pastel-ui'

function Loading(props){
  return (
    <Overlay>
      <Icon name="loading" width={30} fill={props.color || colors.secondary} />
    </Overlay>
  )
}

export default Loading