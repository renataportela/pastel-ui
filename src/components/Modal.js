import React, { useEffect, useRef, useState } from 'react';
import ReactDOM from 'react-dom';
import styled from '@emotion/styled';

import { scale } from '~/pastel-ui/theming';
import { Overlay, Paper } from '~/pastel-ui';
import Button from '~/pastel-ui/components/Button';
import useOutsideCloser from '~/pastel-ui/hooks/useOutsideCloser';

const modalRoot = document.getElementById('modal-root');

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 300;

  & > div {
    min-width: 300px;
    max-width: 90%;
    animation: ${scale} .1s;

    @media (min-width: 778px) {
      min-width: 600px;
    }
  }
`;

const CloseContainer = styled.div`
  float: right;
  margin-left: 20px;
  margin-bottom: 20px;
`;

const CloseButton = styled(Button)`
  position: absolute;
  top: 12px;
  right: 12px;
`;

function Modal(props){
  const [el] = useState(document.createElement('div'));
  const modalBoxRef = useRef();
  const close = () => props.setOpen(false);

  useOutsideCloser(modalBoxRef, close);

  useEffect(() => {
    modalRoot.appendChild(el);
    return () => modalRoot.removeChild(el);
  }, []);  

  if (props.open){
    return ReactDOM.createPortal(
        <Overlay>
          <Centered>
            <Paper ref={modalBoxRef}>
              <CloseContainer>
                <CloseButton 
                  pallete="primary" 
                  variant="outline" 
                  title={props.titleClose || 'Close'}
                  icon="close"
                  round
                  onClick={close}
                />
              </CloseContainer>
              {props.children}
            </Paper>
          </Centered>
        </Overlay>,
      el
    );
  }

  return null;
}

export default Modal;