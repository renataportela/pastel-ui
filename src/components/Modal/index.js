export { default } from './Modal'
export { default as ModalBase } from './ModalBase'