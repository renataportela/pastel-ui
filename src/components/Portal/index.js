export { default } from './Portal'
export { prepareClientPortals } from '@jesstelford/react-portal-universal'