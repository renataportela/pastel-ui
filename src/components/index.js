// Content
export { default as Avatar } from './content/Avatar'
export { default as Badge } from './content/Badge'
export { default as Chip } from './content/Chip'
export { default as Icon, iconNames, Svg } from './content/Icon'
export { default as Tag } from './content/Tag'
export { Heading, Paragraph, Text } from './content/typography'

// Data Entry
export { default as TextField } from './dataEntry/TextField'

// Dialog
export { default as Alert } from './dialog/Alert'

// Layout
export * from './layout/Grid'
export { default as BackTop } from './layout/BackTop'
export { default as Button, Close } from './layout/Button'
export { default as Ripple } from './layout/Ripple'
export { default as Theme, BaseColors, useTheme } from './layout/Theme'