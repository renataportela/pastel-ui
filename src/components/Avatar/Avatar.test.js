import React from 'react'
import renderer from 'react-test-renderer'

import { Theme, Avatar } from '~/components'

describe('Avatar component', () => {
  it('renders a small Avatar with AL initials and secondary color', () => {
    const tree = renderer.create(
      <Theme>
        <Avatar size="sm" text="Aleks" color="secondary" />
      </Theme>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a medium Avatar with RE initials and warning color', () => {
    const tree = renderer.create(
      <Theme>
        <Avatar size="md" text="Renata" color="warning" />
      </Theme>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a large Avatar with user icon and info color', () => {
    const tree = renderer.create(
      <Theme>
      <Avatar size="lg" icon="user" color="info" />      
      </Theme>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an xLarge Avatar with an image', () => {
    const tree = renderer.create(
      <Theme>
        <Avatar size="xl" text="Alternative" image="https://image.freepik.com/free-vector/cute-panda-waving-hand-icon-illustration-panda-mascot-cartoon-character-animal-icon-concept-isolated_138676-843.jpg" />
      </Theme>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})