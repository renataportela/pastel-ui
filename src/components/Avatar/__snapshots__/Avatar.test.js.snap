// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders a large Avatar with user icon and info color 1`] = `
<span
  className="BaseColors__BaseStyle-s1fbie-0 dZhYCv Avatar__Round-sc-17gj85m-0 koRsku"
  size="lg"
>
  <svg
    className="Svg__SvgStyle-pyey0i-0 gYARyh"
    fill="none"
    height="24px"
    size={null}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
    />
    <circle
      cx="12"
      cy="7"
      r="4"
    />
  </svg>
</span>
`;

exports[`Avatar component renders a medium Avatar with RE initials and warning color 1`] = `
<span
  className="BaseColors__BaseStyle-s1fbie-0 jTDUIl Avatar__Round-sc-17gj85m-0 dfehan"
  size="md"
  title="Renata"
>
  <span
    className="Avatar__Text-sc-17gj85m-2 cFsSmv"
  >
    Re
  </span>
</span>
`;

exports[`Avatar component renders a small Avatar with AL initials and secondary color 1`] = `
<span
  className="BaseColors__BaseStyle-s1fbie-0 dqdQsy Avatar__Round-sc-17gj85m-0 bjPCmA"
  size="sm"
  title="Aleks"
>
  <span
    className="Avatar__Text-sc-17gj85m-2 cFsSmv"
  >
    Al
  </span>
</span>
`;

exports[`Avatar component renders an xLarge Avatar with an image 1`] = `
<span
  className="BaseColors__BaseStyle-s1fbie-0 cKbdWp Avatar__Round-sc-17gj85m-0 gTkBpa"
  size="xl"
  title="Alternative"
>
  <img
    className="Avatar__Image-sc-17gj85m-1 ftoiXU"
    src="https://image.freepik.com/free-vector/cute-panda-waving-hand-icon-illustration-panda-mascot-cartoon-character-animal-icon-concept-isolated_138676-843.jpg"
  />
  <span
    className="Avatar__Text-sc-17gj85m-2 cFsSmv"
  >
    Al
  </span>
</span>
`;
